### PR TITLE
Avoid allocation of Reporter.ERROR to determine !hasErrors

### PR DESCRIPTION
**Pull request by retronym**

### CODE REVIEW

**Comment by som-snytt** [comment]
This sounds [familiar](https://github.com/scala/scala/blob/2.13.x/src/compiler/scala/tools/nsc/reporters/DisplayReporter.scala#L180).

Is this change for the context reporters? I noticed them again today; while trying to update my PR for context lookup, I discovered it was refactored and had to read it again.

I don't remember why the new reporter isn't used by default yet. But I also don't remember asking how context reporters fit in.

File: src/reflect/scala/reflect/internal/Reporting.scala [hunk 1]
```diff
@@ -94,15 +94,18 @@ abstract class Reporter {
...
-  object ERROR   extends Severity(2)("ERROR")
+  private[this] var isERRORInitialized = false
+  object ERROR   extends Severity(2)("ERROR") {
+    isERRORInitialized = true
+  }
 
   def count(severity: Severity): Int       = severity.count
   def resetCount(severity: Severity): Unit = severity.count = 0
 
   def errorCount: Int   = count(ERROR)
```
Comment (som-snytt) [inline src/reflect/scala/reflect/internal/Reporting.scala]: Asking for errorCount would init ERROR. Maybe set the flag from error and reset from reset? I think that's the intention in the filtering reporter, namely, we have an error when an ERROR is actually passed, not just when it's issued. "The ERROR has landed."

But setting the flag from the initializer is pretty clever. I'll try to refresh my memory about reporters.

**Review by som-snytt — APPROVED** [review]
This has the peculiar advantage of setting the flag once.

Approving if approval is required by midnight Brisbane time. Is it summer or winter? Wet or dry season? There were just raindrops on the roof and I checked weather.com to verify.

File: src/reflect/scala/reflect/internal/Reporting.scala (hunk 1 above)
Comment (retronym) [reply src/reflect/scala/reflect/internal/Reporting.scala]: Looks like `errorCount` is only called under `-Xmaxerrs` / `-Xmaxwarns`, which isn't exercised under my profile setup. I might as well add the check for `isERRORInitialized`.

**Comment by retronym** [comment]
>  Is it summer or winter?

Summer 2.0. 😓 🔥

**Pull request merged**